core/templates/dev/head/domain/classifier/ClassifierObjectFactorySpec.js

```diff
@@ -0,0 +1,38 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the ClassifierObjectFactory.
+ */
+ 
+describe('Classifier Object Factory', function() {
+  var ClassifierObjectFactory;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($injector) {
+    ClassifierObjectFactory = $injector.get('ClassifierObjectFactory');
+  }));
+
+  it('should create a new classifier', function() {
+    var classifierObject = (
+     ClassifierObjectFactory.create(
+        0, 'classifierData', 'dataSchemaVersion'));
```

Comment:
Try seeing where this is actually called from, and maybe simulate a call and do some console logging. I think the types of the fields are wrong, e.g. I suspect classifierData isn't a string. If you want to help add some lightweight docs to the fields in the ClassifierObjectFactory file to help future devs not need to rediscover what you find, that would be useful too, but that's optional.

Other than that, the structure of this test looks good. Thanks!

Reply:
OK sir

Reply:
Sir how to check what is type of the field ?

Reply:
Sir i check this file , if i change the type of any fields then it failed the frontend test

Reply:
This one is actually a little difficult to test manually, since it refers to classifiers which aren't actually live in production yet. So I suggest tracing the args: search for where ClassifierObjectFactory is imported in the codebase, and then look at what imports that, etc. until you find a place where you can console.log the stuff passed to it and can verify the types. You'll need to know how to search and trace the path of code through a codebase, and I suggest asking @shubha1593 @DubeySandeep or @kevinlee12 if you need help.

Reply:
Ok i'll try

Reply:
@seanlip sir please take look in changes